src/uPlot.js

```diff
@@ -1658,29 +1658,38 @@ export default function uPlot(opts, data, then) {
 			}
 			else {
 				// setSelect should not be triggered on move events
-				let uni = drag.uni;
+				let dx = abs(mouseLeft0 - mouseLeft1);
+				let dy = abs(mouseTop0 - mouseTop1);
 
-				if (uni != null) {
-					let dx = abs(mouseLeft0 - mouseLeft1);
-					let dy = abs(mouseTop0 - mouseTop1);
+				let dragX = drag.x && dx >= drag.dist;
+				let dragY = drag.y && dy >= drag.dist;
 
-					dragX = dx >= uni;
-					dragY = dy >= uni;
+				let uni = drag.uni;
 
-					// force unidirectionality when both are under uni limit
-					if (!dragX && !dragY) {
-						if (dy > dx)
-							dragY = true;
-						else
-							dragX = true;
+				if (uni != null) {
+					// only calc drag status if they pass the dist thresh
+					if (dragX && dragY) {
+						dragX = dx >= uni;
+						dragY = dy >= uni;
+	
+						// force unidirectionality when both are under uni limit
+						if (!dragX && !dragY) {
+							if (dy > dx)
+								dragY = true;
+							else
+								dragX = true;
+						}
 					}
+				} else if (drag.x && drag.y && dragX != dragY) {
```

Comment:
you dont need to change this here (i can do it as a follow-up) but for future reference, i don't like cuddled `else` blocks.

Reply:
i know this is a bit less efficient, but to me `&& (dragX || dragY)` would be a bit more clear.

Reply:
@EmiPhil i see `dragX != dragY` is still in place. was there an issue with `(dragX || dragY)`, or is this still WIP?